Filter/AbstractDateFilter.php

```diff
@@ -18,18 +18,9 @@
 abstract class AbstractDateFilter extends Filter
 {
     /**
-     * Flag indicating that filter will have range.
-     *
-     * @var bool
-     */
-    protected $range = false;
-
-    /**
-     * Flag indicating that filter will filter by datetime instead by date.
-     *
-     * @var bool
+     * @return string
      */
-    protected $time = false;
+    abstract protected function getFilterTypeClass();
```

Comment:
Adding an abstract protected method to a non-final class is a BC-break for extending classes.